templates/base/client/.hathora/client.ts.hbs

```diff
@@ -127,13 +127,21 @@ export class HathoraConnection {
   }
 
   public async connect(): Promise<void> {
-    this.transport = await this._client.connect(
-      this.token,
-      this.stateId,
-      this.handleData,
-      this.handleClose,
-      this.transportType
-    );
+    return new Promise(async (resolve, reject) => {
+      this.transport = await this._client.connect(
+        this.token,
+        this.stateId,
+        (data) => {
+          resolve();
```

Comment:
will this keep calling resolve() on every message? I think it doesn't hurt but seems unnecessary. 

I'm approving given it's not important to fix

Reply:
yeah resolve/reject won't do anything if it's called after the Promise has already been resolved/rejected. i think this is a normal pattern